modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/internal/upgrade/v2_5_0/UpgradeFragmentEntry.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.fragment.internal.upgrade.v2_5_0;
+
+import com.liferay.fragment.internal.upgrade.v2_5_0.util.FragmentEntryTable;
+import com.liferay.fragment.model.FragmentEntry;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.dao.orm.common.SQLTransformer;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.sql.Statement;
+
+/**
+ * @author Rubén Pulido
+ */
+public class UpgradeFragmentEntry extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		upgradeSchema();
+		upgradeFragmentEntryHeadIdAndHead();
+		upgradeFragmentEntryCounter();
+	}
+
+	protected void upgradeFragmentEntryCounter() throws Exception {
+		runSQL(
+			StringBundler.concat(
+				"insert into Counter (name, currentId) select '",
+				FragmentEntry.class.getName(),
+				"', max(fragmentEntryId) from FragmentEntry"));
+	}
+
+	protected void upgradeFragmentEntryHeadIdAndHead() throws Exception {
+		try (Statement s = connection.createStatement()) {
+			StringBundler sb = new StringBundler(3);
+
+			sb.append("update FragmentEntry set headId = - fragmentEntryId, ");
+			sb.append("head = [$TRUE$] where status = ");
+			sb.append(WorkflowConstants.STATUS_APPROVED);
```

Comment:
Also need to update the status != WorkflowConstants.STATUS_APPROVED rows.

Reply:
Thanks for catching this @Preston-Crary!

Reply:
Fixed [here](https://github.com/brianchandotcom/liferay-portal/pull/90187)